src/inmanta/env.py

```diff
@@ -1177,16 +1177,16 @@ def notify_change(self) -> None:
 
 
 @stable_api
-def mock_process_env(*, python_path: Optional[str] = None, env_path: Optional[str] = None) -> None:
+def mock_process_env(new_process_env: ActiveEnv) -> None:
     """
     Overrides the process environment information. This forcefully sets the environment that is recognized as the outer Python
     environment. This function should only be called when a Python environment has been set up dynamically and this environment
     should be treated as if this process was spawned from it, and even then with great care.
 
-    :param python_path: The path to the python binary. Only one of `python_path` and `env_path` should be set.
-    :param env_path: The path to the python environment directory. Only one of `python_path` and `env_path` should be set.
+    :param new_process_env: new env to use
     """
-    process_env.__init__(python_path=python_path, env_path=env_path)  # type: ignore
+    global process_env
+    process_env = new_process_env
```

Comment:
I simplified this, I hope it is OK

Reply:
@sanderr do you think this is OK?

Reply:
I'm not sure it is. I'd have to check where exactly we use it, but I believe this will break any non-qualified imports. We could have explicitly disallowed it, but I don't think we did. We may be lucky in that all our uses are qualified, in which case perhaps we can get away with it.